daemon/cleanup.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/cilium/cilium/pkg/pidfile"
+)
+
+func handleInterrupt() <-chan struct{} {
```

Comment:
This might be better names as `handleSignals` or `notifyOnExitSignal` (or something), since it's really a one-time way to signal that an exit signal has been seen and the cleanup is complete.